qemu/tests/balloon_boot_in_pause.py

```diff
@@ -0,0 +1,233 @@
+        return (mmem, gmem)
+
+    @error_context.context_aware
+    def balloon_memory(self, new_mem):
+        """
+        Baloon guest memory to new_mem
+
+        :param new_mem: New desired memory.
+        :type new_mem: int
+        """
+        error_context.context("Change VM memory to %s" % new_mem, logging.info)
+        try:
+            self.vm.balloon(new_mem)
+        except Exception, e:
+            if self.vm.monitor.verify_status('paused'):
+                # Make sure memory not changed before the guest resumed
+                if self.get_ballooned_memory() != self.ori_mem:
+                    raise exceptions.TestFail("Memory changed before guest "
+                                              "resumed")
+                logging.info("Resume the guest")
+                self.vm.resume()
+            elif new_mem == self.get_ballooned_memory():
+                pass
+            else:
+                raise exceptions.TestFail("Balloon memory fail with error"
+                                          " message: %s" % e)
+        compare_mem = new_mem
+        balloon_timeout = float(self.params.get("balloon_timeout", 100))
+        status = utils_misc.wait_for((lambda: compare_mem ==
+                                      self.get_ballooned_memory()),
+                                     balloon_timeout)
+        if status is None:
+            raise exceptions.TestFail("Failed to balloon memory to expect"
+                                      " value during %ss" % balloon_timeout)
+
+    def get_memory_boundary(self):
+        """
+        Get the legal memory boundary for the balloon test
+
+        :return: min and max size of the memory
+        :rtype: tuple
+        """
+        if self.ori_mem <= (1 * 1024 * 1024):
+            ratio = float(self.params.get("ratio_1", 0.8))
+        else:
+            ratio = float(self.params.get("ratio_2", 0.5))
+        min_size = int(ratio * self.ori_mem)
+        max_size = self.ori_mem
+        return min_size, max_size
+
+    def error_report(self, step, expect_value, monitor_value, guest_value):
+        """
+        Generate the error report
+
+        :param step: the step of the error happen
+        :param expect_value: memory size assign to the vm
+        :param monitor_value: memory size report from monitor
+        :param guest_value: memory size report from guest
+        """
+        logging.error("Memory size mismatch %s:\n" % step)
+        error_msg = "Wanted to be changed: %s\n" % abs(self.old_mmem -
+                                                       expect_value)
+        error_msg += "Changed in monitor: %s\n" % abs(self.old_mmem -
+                                                      monitor_value)
+        if self.old_gmem:
+            error_msg += "Changed in guest: %s\n" % abs(
+                self.old_gmem - guest_value)
+        logging.error(error_msg)
+
+
+class BallooningTestPauseWin(BallooningTestPause):
```

Comment:
@CongLi function get_memory_status() is defined in subclasses BallooningTestWin() and BallooningTestLinux() respectively, while not in the base class BallooningTest(). So if I don't define them again, I could not use the function in my script since I only inherit the base class BallooningTest() in my class BallooningTestPause()